### ISSUE TITLE
Procedures and GetProcedures both public

### ISSUE THREAD

**Issue by AFract** [ErikEJ/EFCorePowerTools#2158]
Hello, 

I have a small question regarding the generated code for a DbContext generated from a SQL Server 2019 database, with stored procedures. This is really a minor issue.

The generated code contains:
`
    public partial class XXDbContext
    {
        private IXXDbContextProcedures _procedures;

        public virtual IXXDbContextProcedures Procedures
        {
            get
            {
                if (_procedures is null) _procedures = new XXDbContextProcedures(this);
                return _procedures;
            }
            set
            {
                _procedures = value;
            }
        }

        public IXXDbContextProcedures GetProcedures()
        {
            return Procedures;
        }
`

**Procedures** and **GetProcedures()** are both public, and both seem to return exactly the same thing (at least in this version). It's easy to figure that as the code is available, but I was wondering why both were existing. 

Also (and this is the reason why I post this issue), it makes Sonar complain with the following violation (S4059) : 
Warning	S4059	Change either the name of property 'Procedures' or the name of method 'GetProcedures' to make them distinguishable.

This is not a huge disturbance, but as it is a generated file there's no simple way to ignore it, apart change the file everytime to add a #pragma warning disable S4059 for example, or tweak the Sonar configuration.
Do you have any advice ? Maybe something that could be added automatically in the generated code to make Sonar ignore it ?

### Provide technical details

- EF Core Power Tools version: 2.6.140
- Exact Visual Studio version:  Visual Studio 2022 17.8.6
- Database engine: SQL Server 2019
- EF Core : EF Core 7
- Is Handlebars templates used: no
- Is T4 templates used: yes (default settings of EFCore PT)
- Is a SQL Server .dacpac used: no

Thank you !

**Comment by ErikEJ** [comment]
removing either will be a breaking change for existing users - but Sonar should ignore the file as it has an autogenerated header

**Comment by AFract** [comment]
You're right about this of course. Regarding <auto-generated>, I don't find it as a documented tag in MS documentation with a quick search. Maybe it's more a convention than a norm ? 
I have tried with both // and /// comment block, just in case this it behaves the same. Anyway, this is probably more a question for Sonar team rather than EFCorePT, since the tag is present.

**Comment by ErikEJ** [comment]
It is mentioned in great detail here: https://github.com/dotnet/roslyn/issues/39758

**Comment by ErikEJ** [comment]
Found this Sonar issue actually: https://community.sonarsource.com/t/sonarqube-does-not-exclude-files-that-are-auto-generated-during-the-dotnet-build-process/80935/4

**Comment by ErikEJ** [comment]
@AFract OK if I close this?
